data_steward/cleaners/clean_rule_4.py

```diff
@@ -0,0 +1,52 @@
+"""
+Rule: 4
+ID columns in each domain should be unique
+"""
+
+import bq_utils
+import cdm
+import clean_cdr_engine
+import resources
+
+rule_4_query = """
+select {columns}
+from (select m.*,
+ROW_NUMBER() OVER (PARTITION BY m.{domain_table}_id) AS row_num
+from `{project_id}.{dataset_id}.{table_name}` as m) as t
+where row_num = 1
+"""
+
+
+def run_clean_rule_4(project_id, dataset_id):
```

Comment:
Names are important.  `run_clean_rule_4` is vague and doesn't provide good information about what is happening.  Try something more descriptive.  Like `run_ensure_unique_ids`.